File: 03_21.02_klassikomplekt/kymnev6istlus/src/main/java/ee/tormi/kymnev6istlus/entity/Points.java
package ee.tormi.kymnev6istlus.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity

public class Points {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long point_id;

    private Double hundredMeterRun;
    private Double longJump;
    private Double shotPut;
    private Double highJump;
    private Double fourHundredMeterRun;
    private Double hundredTenMeterHurdle;
    private Double discusThrow;
    private Double poleVault;
    private Double javelin;
    private Double thousandFiveHundredMeterRun;

    private Double totalScore;

    @ManyToOne (fetch = FetchType.LAZY)
    @JoinColumn (name = "athlete_id", nullable = false)
    private Athlete athlete;
}

File: 03_21.02_klassikomplekt/FEkymnev6istlus/src/models/Points.ts
import type { Athlete } from "./Athletes"

export type Points = {
    pointId: number,
    hundredMeterRun: number,
    longJump: number,
    shotPut: number,
    highJump: number,
    fourHundredMeterRun: number,
    hundredTenMeterHurdle: number,
    discusThrow: number,
    poleVault: number,
    javelin: number,
    thousandFiveHundredMeterRun: number,

    totalScore: number,

    athlete: Athlete
}